### DISCUSSION TITLE
Fortnite Discovery V2 "Invalid security token"

### DISCUSSION THREAD

**Discussion by DrizzleMcFizzle** [LeleDerGrasshalmi/FortniteEndpointsDocumentation#290] [Q&A] [answered]
I've been trying to scrape the discovery data, I managed to get it working with v1 a while back, but cannot get it to work for the life of me with v2. I've ensured that the x-epic-access-token is correct, here's my request: 

```
x_epic_access_token = "fL2LFlLhqhUTTGOK1x84t+7F1aMwT6uBYBJ+2McAzaY="
url = f"https://fn-service-discovery-live-public.ogs.live.on.epicgames.com/api/v2/discovery/surface/{surface_name}"

headers = {
    "User-Agent": "Fortnite/++Fortnite+Release-30.20-CL-26171015 Windows/10",
    "Authorization": f"Bearer {access_token}",
    "X-Epic-Access-Token": x_epic_access_token,
    "Content-Type": "application/json"
}

data = {
    "playerId": account_id,
    "partyMemberIds": [account_id],
    "platform": "Windows",
}

params = {
    "appId": "Fortnite",
    "stream": "%2B%2BFortnite%2BRelease-30.20"
}
```

Response:
`{'errorCode': 'errors.com.epicgames.failed_surface_token_validation', 'errorMessage': 'Invalid security token.'}`

Cheers!

**Comment by LeleDerGrasshalmi** [comment, marked as answer]
are you sure your python lib doesnt again url encode the query params? i'd assume thats whats happening, so epic doesnt see it correct

**Reply by DrizzleMcFizzle** [reply]
I can't believe it was just this the whole time... cheers

**Reply by LeleDerGrasshalmi** [reply]
np
